examples/nodejs/index.js

```diff
@@ -0,0 +1,289 @@
+
+  // [START alloydb_knex_backoff]
+  // 'knex' uses a built-in retry strategy which does not implement backoff.
+  // 'createRetryIntervalMillis' is how long to idle after failed connection creation before trying again
+  config.pool.createRetryIntervalMillis = 200; // 0.2 seconds
+  // [END alloydb_knex_backoff]
+
+  // Check if a Secret Manager secret version is defined
+  // If a version is defined, retrieve the secret from Secret Manager and set as the DB_PASS
+  const {ALLOYDB_CREDENTIALS_SECRET} = process.env;
+  if (ALLOYDB_CREDENTIALS_SECRET) {
+    const secrets = await accessSecretVersion(ALLOYDB_CREDENTIALS_SECRET);
+    try {
+      process.env.DB_PASS = secrets.toString();
+    } catch (err) {
+      err.message = `Unable to parse secret from Secret Manager. Make sure that the secret is JSON formatted: \n ${err.message} `;
+      throw err;
+    }
+  }
+  // Use a TCP socket.
+  return createTcpPool(config);
+};
+
+const ensureSchema = async pool => {
+  const hasTable = await pool.schema.hasTable('votes');
+  if (!hasTable) {
+    return pool.schema.createTable('votes', table => {
+      table.increments('vote_id').primary();
+      table.timestamp('time_cast', 30).notNullable();
+      table.specificType('candidate', 'CHAR(6)').notNullable();
+    });
+  }
+  logger.info("Ensured that table 'votes' exists");
+};
+
+const createPoolAndEnsureSchema = async () =>
+  await createPool()
+    .then(async pool => {
+      await ensureSchema(pool);
+      return pool;
+    })
+    .catch(err => {
+      logger.error(err);
+      throw err;
+    });
+
+// [START alloydb_knex_connection]
+/**
+ * Insert a vote record into the database.
+ *
+ * @param {object} pool The Knex connection object.
+ * @param {object} vote The vote record to insert.
+ * @returns {Promise}
+ */
+const insertVote = async (pool, vote) => {
+  try {
+    return await pool('votes').insert(vote);
+  } catch (err) {
+    throw Error(err);
+  }
+};
+// [END alloydb_knex_connection]
+
+/**
+ * Retrieve the latest 5 vote records from the database.
+ *
+ * @param {object} pool The Knex connection object.
+ * @returns {Promise}
+ */
+const getVotes = async pool => {
+  return await pool
+    .select('candidate', 'time_cast')
+    .from('votes')
+    .orderBy('time_cast', 'desc')
+    .limit(5);
+};
+
+/**
+ * Retrieve the total count of records for a given candidate
+ * from the database.
+ *
+ * @param {object} pool The Knex connection object.
+ * @param {object} candidate The candidate for which to get the total vote count
+ * @returns {Promise}
+ */
+const getVoteCount = async (pool, candidate) => {
+  return await pool('votes').count('vote_id').where('candidate', candidate);
+};
+
+const httpGet = async (req, res) => {
+  pool = pool || (await createPoolAndEnsureSchema());
+  try {
+    // Query the total count of "TABS" from the database.
+    const tabsResult = await getVoteCount(pool, 'TABS');
+    const tabsTotalVotes = parseInt(tabsResult[0].count);
```

Comment:
isn't `count` already a number? are you sure it needs [`parseInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt) here? `parseInt` is only meant to convert strings to number, if that's the intent you might still want to specify a `radix` value as better explained in the doc linked here.